### PR TITLE
Remove pytest _describe

### PR DESCRIPTION
**Pull request by mashrikt**
This refers to #4270. Removes the pytest_describe blocks with class.
I am told that it is blocked until https://github.com/orgs/rtfd/projects/2 is resolved.

### CODE REVIEW

File: readthedocs/config/tests/test_config.py [hunk 1]
```diff
@@ -222,15 +222,15 @@ def test_python_pip_install_default():
     assert build.pip_install is False
 
 
-def describe_validate_python_extra_requirements():
+class ValidatePythonExtraRequirements:
```
Comment (stsewd) [inline readthedocs/config/tests/test_config.py]: We need to follow the convention of naming the class starting with `Test` like `TestValidatePythonExtraRequirements` and inheritance from object to keep the code python2 compatible.

File: readthedocs/config/tests/test_config.py [hunk 2]
```diff
@@ -222,15 +222,15 @@ def test_python_pip_install_default():
...
 
-    def it_defaults_to_list():
+    def it_defaults_to_list(self):
```
Comment (stsewd) [inline readthedocs/config/tests/test_config.py]: Also in the methods, we need to start the name with `test_`

Comment (mashrikt) [reply readthedocs/config/tests/test_config.py]: Right! My bad.

Comment (mashrikt) [reply readthedocs/config/tests/test_config.py]: should I use test_it_defaults_to_list or test_defaults_to_list?

**Comment by mashrikt** [comment]
https://github.com/rtfd/readthedocs.org/pull/4355/files is merged.
So I have rebased this PR. But checks have failed.
@stsewd

**Comment by safwanrahman** [comment]
@mashrikt I have restarted the test!

**Comment by stsewd** [comment]
@mashrikt could you rebase/update the branch?

**Comment by mashrikt** [comment]
@stsewd Done!

File: readthedocs/config/tests/test_config.py [hunk 3]
```diff
@@ -274,15 +274,15 @@ def test_python_pip_install_default():
     assert build.python.install_with_pip is False
 
 
-def describe_validate_python_extra_requirements():
+class TestValidatePythonExtraRequirements(object):
 
-    def it_defaults_to_list():
+    def test_it_defaults_to_list(self):
```
Comment (stsewd) [inline readthedocs/config/tests/test_config.py]: not sure if we need the `it` in the tests names anymore, what do you think?

Comment (stsewd) [reply readthedocs/config/tests/test_config.py]: I'm not an english expert, but I guess 2 makes sense.

**Comment by ericholscher** [comment]
@stsewd want to give this another look and see if we can get it approved and merged?

File: readthedocs/config/tests/test_config.py (hunk 3 above)
Comment (ericholscher) [reply readthedocs/config/tests/test_config.py]: Yea, 2 is 👍

**Comment by ericholscher** [comment]
As a note, I'm happy to merge it with the current naming, so if everything looks good, we should 👍

**Comment by stsewd** [comment]
@ericholscher yeah, we are good to merge this, codecov bot didn't got triggered here :/

**Pull request merged**